src/librustc_resolve/lib.rs

```diff
@@ -2793,11 +2790,9 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
 
         // Move back up.
         self.current_module = orig_module;
-        for _ in 0 .. num_macro_definition_ribs {
-            self.ribs[ValueNS].pop();
-            self.label_ribs.pop();
-        }
-        self.ribs[ValueNS].pop();
+        let valuens_len = self.ribs[ValueNS].len();
+        self.ribs[ValueNS].truncate(valuens_len - num_macro_definition_ribs - 1);
```

Comment:
hm, why is this offset by 1 but the next one not?

Reply:
This preserves the existing logic; `self.ribs[ValueNS]` was popped once more after the loop.

Reply:
Please, keep the old obvious logic (`pop`s are symmetric with pushes above), unless you have benchmarks.

Reply:
@petrochenkov `truncating` is [faster](https://gist.github.com/ljedrz/ca338bd346fb870c5d808d15969208ea) - the more elements to `pop`, the faster it becomes.

Reply:
I know that it's faster, I meant that it doesn't matter that it's faster, because the speed up is unnoticeable in the general picture. In this case readability/simplicity is very much preferable.

(By benchmarks I mean e.g. benchmark showing that this specific code is hot and takes a significant percent of resolve time.)

Reply:
Unfortunately I don't have a Linux rig to do perf runs :/. I can revert to the original code if you don't think it is hot enough.